### PR TITLE
mech bustin update 2022

### LINKED ISSUE #70845: Even if you have no arms and no legs you can still enter a mech and then you get stuck inside because you can't eject
## Testmerges:

i don't think someone with no limbs should be able to enter a mech, and preferably just anyone with no arms as they can't pilot it and will be stuck

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Adds a huge ass crowbar to robotics (the mech removal tool), it deals 5 damage unwielded, or 19 wielded. (should be fine, considering robotics also has the easiest access to the materials needed for a chainsaw)
You can use it while wielded on mechs to break the occupants out. This takes 5 seconds (or 3 in an unenclosed mech like a ripley)
When you die in a mech you no longer automatically get ejected.
refactors fire axe cabinets to support more items than the fireaxe
makes some vehicle code better
closes #70845 (you can still enter a mech without limbs, i think thats fine because you can use it to protect yourself from death in a dangerous situation or something until someone breaks you out with the really large crowbar)
video: https://streamable.com/x4gom2

## Why It's Good For The Game

robotics having a giant ass crowbar to break people out of mechs seems like a fun idea
you currently cant exit a mech if youre incapacitated inside it unless you DIE

## Changelog

:cl: Fikou, sprites by Halcyon
refactor: fire axe cabinets support items that aren't fire axes
balance: mechs no longer eject you when you die in them
add: Adds a giant crowbar to robotics, it can break open mechs to eject their pilots.
/:cl:

### CODE REVIEW

**Comment by TheBoondock** [comment]
should security gets one in  their department as well, since if a mech gone rouge they still have a crowbar to pry it open

File: code/game/objects/items/tools/crowbar.dm [hunk 1]
```diff
@@ -3,7 +3,6 @@
 	desc = "A small crowbar. This handy tool is useful for lots of things, such as prying floor tiles or opening unpowered doors."
 	icon = 'icons/obj/tools.dmi'
 	icon_state = "crowbar"
-	inhand_icon_state = "crowbar"
```
Comment (ShizCalev) [inline code/game/objects/items/tools/crowbar.dm:6]: Be explicit with inhand icons that exist please, removing this removes the ability to unit test for missing icons.

Override subtypes that shouldn't have them with `inhand_icon_state = null` instead.

File: code/game/objects/items/tools/crowbar.dm [hunk 2]
```diff
@@ -31,7 +30,6 @@
 
 /obj/item/crowbar/red
 	icon_state = "crowbar_red"
-	inhand_icon_state = "crowbar_red"
```
Comment (ShizCalev) [inline code/game/objects/items/tools/crowbar.dm:34]: Same.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ShizCalev** [comment]
Rest of the code looks good 👍

File: code/game/objects/structures/fireaxe.dm [hunk 3]
```diff
@@ -10,62 +10,69 @@
...
 			broken = FALSE
 			atom_integrity = max_integrity
 			update_appearance()
 	else if(open || broken)
-		if(istype(I, /obj/item/fireaxe) && !fireaxe)
-			if(HAS_TRAIT(I, TRAIT_WIELDED))
-				to_chat(user, span_warning("Unwield [I] first."))
+		if(istype(attacking_item, item_path) && !held_item)
+			if(HAS_TRAIT(attacking_item, TRAIT_WIELDED))
+				to_chat(user, span_warning("Unwield [attacking_item] first."))
```
Comment (ShizCalev) [inline code/game/objects/structures/fireaxe.dm:57]: Since this is a failure state, should it end with `!`?

Comment (Fikou) [reply code/game/objects/structures/fireaxe.dm:57]: sure

File: code/game/objects/items/tools/crowbar.dm (hunk 1 above)
Comment (Fikou) [reply code/game/objects/items/tools/crowbar.dm:6]: this breaks the inhands for the crowbar

**Comment by moocowswag** [comment]
If the mech turns does it stop the do_after?

It would be weird that youre able to continue prying on a durand thats spinning around in place.

**Comment by carshalash** [comment]
You call this a bustin PR, but you didn't play the ghostbusters theme in the demonstration video? For shame fikou.

**Review by ShizCalev — APPROVED** [review]

**Comment by newfren** [comment]
no sprite change on mob between wielded and unwielded?

**Comment by Fikou** [comment]
> no sprite change on mob between wielded and unwielded?

thats not the case it was just bugged

**Comment by Shadowflame909** [comment]
Nanotransen has upped their game with mech sized crowbar removers

Will the syndicate counteract with an even faster red mech sized crowbar? Maybe an anti-crowbar removal device? Maybe a bluespace mech remover crowbar?

We will see eventually

**Comment by improvedname** [comment]
>Mech made of reinforced alloy knowing resistant to bullets high force brute damage and explosions

>pried open by a comically large crowbar

**Comment by ATH1909** [comment]
should probably add the ability to do this to jaws of life as well, since they're used to extract people from car wrecks irl

and maybe to the fire closet crowbars too, since they're also   l a r g e ?

**Comment by ATH1909** [comment]
5 seconds for the extraction also seems a bit... short?

**Comment by LarissaMayrink** [comment]
5 seconds is too short.

Sec should get one too

and does changing directions cancell someone from removing you?

**Comment by Fikou** [comment]
> should probably add the ability to do this to jaws of life as well, since they're used to extract people from car wrecks irl

no they aren't large enough

> and maybe to the fire closet crowbars too, since they're also l a r g e ?

no fuck you they are tiny as shit

> 5 seconds is too short.

not really, its a specialized tool, theres only one on station, if you cant deal with that in 5 seconds or move away thats your problem

> Sec should get one too

i disagree

> and does changing directions cancell someone from removing you?

ill be adding that now

**Comment by GuillaumePrata** [comment]
Roboticists can actually stop other shitter roboticists properly now.

You know what they say, the only way to stop a Bad Roboticist with a mech is a good roboticist with a mech... prying crowbar!

**Comment by Wallemations** [comment]
Is there any way to tell if someone inside a mech is dead? Some sort of indicator or examine text? Maybe just the mech saying the driver has died on a 30 second to 1 minute timer or something?

**Comment by optimumtact** [comment]
> Is there any way to tell if someone inside a mech is dead? Some sort of indicator or examine text? Maybe just the mech saying the driver has died on a 30 second to 1 minute timer or something?

Go up and talk to them?

**Comment by Wallemations** [comment]
> > Is there any way to tell if someone inside a mech is dead? Some sort of indicator or examine text? Maybe just the mech saying the driver has died on a 30 second to 1 minute timer or something?
> 
> Go up and talk to them?

I mean fair, but some people might just go SSD inside their mechs. Maybe having a do_after when you examine a mech right next to it to have you check if the person inside is alive?

**Comment by Miczu555PL** [comment]
> its a specialized tool, theres only one on station
Fireaxea can be remade in atmos. How about this one?

**Comment by Wallemations** [comment]
Wait there's only one? How are you supposed to get a dead guy out of a mech if it gets lost or destroyed?

**Comment by githubuser4141** [comment]
If you're lagging or have occasional bursts of lag (2-5 secoonds) 5 seconds would be round ending for you. Can it be upped to at least 7 seconds so it's more forgiving for people with latency or if you're not paying attention?

**Comment by Valorium-RB** [comment]
I mean, arguably having 5 seconds of lag can get you killed in a LOT of scenarios, but 5 seconds is pretty short.

**Comment by Redrover1760** [comment]
> I mean, arguably having 5 seconds of lag can get you killed in a LOT of scenarios, but 5 seconds is pretty short.

Especially if you were in a nukie mech. You might be able to take 5 seconds of bullets, but one guy with funny crowbar means you are ejected.

File: code/game/objects/items/fireaxe.dm [hunk 4]
```diff
@@ -18,7 +18,6 @@
 	attack_verb_simple = list("attack", "chop", "cleave", "tear", "lacerate", "cut")
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	sharpness = SHARP_EDGED
-	max_integrity = 200
```
Comment (LemonInTheDark) [inline code/game/objects/items/fireaxe.dm:21]: hm?

File: code/game/objects/items/tools/crowbar.dm [hunk 5]
```diff
@@ -54,7 +54,6 @@
 	throw_range = 3
 	custom_materials = list(/datum/material/iron=70)
 	icon_state = "crowbar_large"
-	inhand_icon_state = "crowbar"
```
Comment (LemonInTheDark) [inline code/game/objects/items/tools/crowbar.dm:57]: also hm?

File: code/game/objects/items/tools/crowbar.dm [hunk 6]
```diff
@@ -162,3 +162,53 @@
...
+	slot_flags = NONE
+	toolspeed = 1.25
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 100, BIO = 0, FIRE = 100, ACID = 0)
+	resistance_flags = FIRE_PROOF
+	bare_wound_bonus = 15
+	wound_bonus = 10
+	/// How much damage to do unwielded
+	var/force_unwielded = 5
+	/// How much damage to do wielded
+	var/force_wielded = 19
```
Comment (LemonInTheDark) [inline code/game/objects/items/tools/crowbar.dm:183]: why are these vars? they don't update on edit

File: code/game/objects/items/tools/crowbar.dm [hunk 7]
```diff
@@ -162,3 +162,53 @@
...
+
+/obj/item/crowbar/mechremoval/Initialize(mapload)
+	. = ..()
+	transform = transform.Translate(0, -8)
```
Comment (LemonInTheDark) [inline code/game/objects/items/tools/crowbar.dm:187]: why pixel shift like this?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/objects/items/fireaxe.dm (hunk 4 above)
Comment (Fikou) [reply code/game/objects/items/fireaxe.dm:21]: 200 is inherited from parent

File: code/game/objects/items/tools/crowbar.dm (hunk 6 above)
Comment (Fikou) [reply code/game/objects/items/tools/crowbar.dm:183]: the fireaxe did it

File: code/game/objects/items/tools/crowbar.dm (hunk 5 above)
Comment (Fikou) [reply code/game/objects/items/tools/crowbar.dm:57]: inherited from parent

File: code/game/objects/items/tools/crowbar.dm (hunk 7 above)
Comment (Fikou) [reply code/game/objects/items/tools/crowbar.dm:187]: pixel changes dont carry over to huds

Comment (Fikou) [reply code/game/objects/items/tools/crowbar.dm:187]: this doesnt carry over to attack anims though but i dont think pixel changes do either

**Comment by KittyNoodle** [comment]
> should security gets one in their department as well, since if a mech gone rouge they still have a crowbar to pry it open (assuming the roboticist is rouge and hid the one in their department)

I believe the typical security response to a rouge mech is to beat the everliving shit out of it

File: code/game/objects/structures/fireaxe.dm [hunk 8]
```diff
@@ -10,62 +10,68 @@
 	integrity_failure = 0.33
 	var/locked = TRUE
 	var/open = FALSE
```
Comment (Fikou) [inline code/game/objects/structures/fireaxe.dm:12]:
```suggestion
	/// Do we need to be unlocked to be opened.
	var/locked = TRUE
	/// Are we opened, can someone take the held item out.
	var/open = FALSE
```

File: code/game/objects/structures/fireaxe.dm [hunk 9]
```diff
@@ -10,62 +10,68 @@
...
-	var/obj/item/fireaxe/fireaxe
+	var/obj/item/held_item
+	var/item_path = /obj/item/fireaxe
+	var/item_overlay = "axe"
```
Comment (Fikou) [inline code/game/objects/structures/fireaxe.dm:15]:
```suggestion
	/// The item we're holding.
	var/obj/item/held_item
	/// The path of the item we spawn and can hold.
	var/item_path = /obj/item/fireaxe
	/// Overlay we get when the item is inside us.
	var/item_overlay = "axe"
```

**Comment by LemonInTheDark** [comment]
Seems cool

**Comment by Nabski89** [comment]
This thing seems to just DESTROY bones,

**Pull request merged**